CMakeLists.txt

```diff
@@ -71,12 +71,21 @@ IF (BUILD_TESTS OR BUILD_EXAMPLES)
     # ##############################################################################
     # Use a header only version of armadillo; default CMake project is not header only
     # That's why we use a custom 'armadillo' target with only include directory
-    #add_subdirectory(third_party/armadillo-code)
-    SET(ARMADILLO_INCLUDE_DIR third_party/armadillo-code/include)
+    if (ARMADILLO_ROOT_DIR)
+        #add_subdirectory(${ARMADILLO_ROOT_DIR} armadillo-code)
+        SET(ARMADILLO_INCLUDE_DIR ${ARMADILLO_ROOT_DIR}/include)
+    else()
+        #add_subdirectory(third_party/armadillo-code)
+        SET(ARMADILLO_INCLUDE_DIR third_party/armadillo-code/include)
+    endif()
     ADD_LIBRARY(armadillo INTERFACE)
     target_include_directories(armadillo INTERFACE ${ARMADILLO_INCLUDE_DIR})
-    
-    add_subdirectory(third_party/pybind11)
+
+    if (PYBIND11_ROOT_DIR)
+        add_subdirectory(${PYBIND11_ROOT_DIR} pybind11)
+    else()
+        add_subdirectory(third_party/pybind11)
+    endif()
 
 
     ENABLE_TESTING()
```

Comment:
Better python3 finder consists in removing line 92 and to use the one found by pybind11 (using `PYTHON_EXECUTABLE` and not `Python3_EXECUTABLE`)